lnrpc-ideal/rpc.proto

```diff
@@ -0,0 +1,809 @@
+syntax = "proto3";
+
+import "google/api/annotations.proto";
+
+package lnrpc;
+
+option go_package = "github.com/lightningnetwork/lnd/lnrpc";
+
+service Lightning {
+    /** lncli: `listinvoices`
+    ListInvoices returns a list of all the invoices currently stored within the
+    database. Any active debug invoices are ignored. It has full support for
+    paginated responses, allowing users to query for specific invoices through
+    their add_index. This can be done by using either the first_index_offset or
+    last_index_offset fields included in the response as the index_offset of the
+    next request. By default, the first 100 invoices created will be returned.
+    Backwards pagination is also supported through the Reversed flag.
+    */
+    rpc ListInvoices(ListInvoiceRequest) returns (ListInvoiceResponse) {
```

Comment:
Part of the `ListInvoices` functionality can also be obtained through `SubscribeInvoices` by passing in a specific add or settle index. 

One question that I have about `ListInvoices` is how to know which snapshot of the invoice the caller got. Invoices are never final, because overpayment is accepted.